src/cmd_all/src/single_node.rs

```diff
@@ -96,50 +139,282 @@ pub fn make_single_node_state_store_url(store_directory: &String) -> String {
+        compactor_opts
+            .opts
+            .insert("--listen-addr".to_string(), compactor_addr.to_string());
+    }
+    if let Some(frontend_node_override_opts) = &opts.frontend_node_override_opts {
+        frontend_opts
+            .opts
+            .extend(frontend_node_override_opts.opts.clone());
     }
+    if let Some(meta_node_override_opts) = &opts.meta_node_override_opts {
+        meta_opts.opts.extend(meta_node_override_opts.opts.clone());
+    }
+    if let Some(compute_node_override_opts) = &opts.compute_node_override_opts {
+        compute_opts
+            .opts
+            .extend(compute_node_override_opts.opts.clone());
+    }
+    if let Some(compactor_node_override_opts) = &opts.compactor_node_override_opts {
+        compactor_opts
+            .opts
+            .extend(compactor_node_override_opts.opts.clone());
+    }
+    let frontend_opts = if opts.disable_frontend {
+        None
+    } else {
+        let args = frontend_opts.raw_opts_to_args();
+        let frontend_opts = FrontendOpts::parse_from(osstrs(args));
+        Some(frontend_opts)
+    };
+    let meta_opts = if opts.disable_meta {
+        None
+    } else {
+        let args = meta_opts.raw_opts_to_args();
+        let meta_opts = MetaNodeOpts::parse_from(osstrs(args));
+        Some(meta_opts)
+    };
+    let compute_opts = if opts.disable_compute {
+        None
+    } else {
+        let args = compute_opts.raw_opts_to_args();
+        let compute_opts = ComputeNodeOpts::parse_from(osstrs(args));
+        Some(compute_opts)
+    };
+    let compactor_opts = if opts.disable_compactor {
+        None
+    } else {
+        let args = compactor_opts.raw_opts_to_args();
+        let compactor_opts = CompactorOpts::parse_from(osstrs(args));
+        Some(compactor_opts)
+    };
     ParsedStandaloneOpts {
-        meta_opts: Some(meta_opts),
-        compute_opts: Some(compute_opts),
-        frontend_opts: Some(frontend_opts),
-        compactor_opts: Some(compactor_opts),
+        frontend_opts,
+        meta_opts,
+        compute_opts,
+        compactor_opts,
+    }
+}
+
+#[derive(Clone, Debug, Eq, PartialEq)]
+struct RawOpts {
+    flags: HashSet<String>,
+    opts: HashMap<String, String>,
+}
+
+// FIXME(kwannoel): This is a placeholder implementation
+impl Ord for RawOpts {
+    fn cmp(&self, other: &Self) -> std::cmp::Ordering {
+        Ordering::Equal
+    }
+}
+
+impl PartialOrd for RawOpts {
+    fn partial_cmp(&self, other: &Self) -> Option<std::cmp::Ordering> {
+        Some(self.cmp(other))
+    }
+}
+
+impl From<&str> for RawOpts {
+    fn from(s: &str) -> Self {
+        let mut opts = HashMap::new();
+        let mut flags = HashSet::new();
+        let args = split(s).unwrap();
+        let mut args = args.into_iter().peekable();
+        loop {
+            let Some(arg) = args.peek() else {
+                break;
+            };
+            assert!(arg.starts_with("--"));
```

Comment:
Short arguments start with `-`

Reply:
`clap` also supports multi-value arguments. The parsing code here looks error-prone.

Reply:
Trade-off is acceptable to me. Later if we start having multi-value args we can also handle them.

Reply:
This section will mostly not be used by users, since it's only exposed to them if they want to change node level opts.
For `single-node` mode, they should be main using `process` level opts.

Reply:
It is meant for the configurable `standalone` mode mainly.